e2e_test.go

```diff
@@ -0,0 +1,1035 @@
+	ctx context.Context
+
+	data []Datum
+
+	insertConfig = &payload.Insert_Config{
+		SkipStrictExistCheck: true,
+	}
+
+	searchNum    = 3
+	searchConfig = &payload.Search_Config{
+		Num:     uint32(searchNum),
+		Radius:  -1.0,
+		Epsilon: 0.1,
+		Timeout: 3000000000,
+	}
+
+	updateConfig = &payload.Update_Config{
+		SkipStrictExistCheck: true,
+	}
+
+	upsertConfig = &payload.Upsert_Config{
+		SkipStrictExistCheck: true,
+	}
+
+	removeConfig = &payload.Remove_Config{
+		SkipStrictExistCheck: true,
+	}
+)
+
+type Datum struct {
+	ID     string    `json:"id"`
+	Vector []float32 `json:"vector"`
+}
+
+func init() {
+	ctx = context.Background()
+
+	err := readJSON()
+	if err != nil {
+		fmt.Errorf("error: %s", err)
+		os.Exit(1)
+	}
+}
+
+func readJSON() error {
+	f, err := os.Open("wordvecs1000.json")
+	if err != nil {
+		return err
+	}
+
+	defer f.Close()
+
+	bytes, err := ioutil.ReadAll(f)
+	if err != nil {
+		return err
+	}
+
+	return json.Unmarshal(bytes, &data)
+}
+
+func beforeEach() {
+	var err error
+
+	conn, err = grpc.DialContext(ctx, addr, grpc.WithInsecure())
+	if err != nil {
+		fmt.Errorf("error: %s", err)
+		os.Exit(1)
+	}
+
+	client = vald.NewValdClient(conn)
+	agentClient = core.NewAgentClient(conn)
+}
+
+func afterEach() {
+	conn.Close()
+}
+
+func TestE2E(t *testing.T) {
+	// Insert
+
+	t.Run(`Test for Insert operation`, func(t *testing.T) {
+		beforeEach()
+		defer afterEach()
+
+		req := &payload.Insert_Request{
+			Vector: &payload.Object_Vector{
+				Id:     data[0].ID,
+				Vector: data[0].Vector,
+			},
+			Config: insertConfig,
+		}
+
+		result, err := client.Insert(ctx, req)
+		if err != nil {
+			t.Fatalf("error: %s", err)
+		}
+
+		if result == nil {
+			t.Fatal("error: returned result is nil")
+		}
+
+		if len(result.GetIps()) != 1 {
+			t.Fatal("error: invalid number of ips")
+		}
+	})
+
+	t.Run(`Test for MultiInsert operation`, func(t *testing.T) {
+		beforeEach()
+		defer afterEach()
+
+		requests := make([]*payload.Insert_Request, 0)
+
+		for i := 1; i < 11; i++ {
+			requests = append(requests, &payload.Insert_Request{
+				Vector: &payload.Object_Vector{
+					Id:     data[i].ID,
+					Vector: data[i].Vector,
+				},
+				Config: insertConfig,
+			})
+		}
+
+		req := &payload.Insert_MultiRequest{
+			Requests: requests,
+		}
+
+		results, err := client.MultiInsert(ctx, req)
+		if err != nil {
+			t.Fatalf("error: %s", err)
+		}
+
+		if results == nil {
+			t.Fatal("error: returned results is nil")
+		}
+
+		for _, result := range results.GetLocations() {
+			if len(result.GetIps()) != 1 {
+				t.Fatal("error: invalid number of ips")
+			}
+		}
+	})
+
+	t.Run(`Test for StreamInsert operation`, func(t *testing.T) {
+		beforeEach()
+		defer afterEach()
+
+		stream, err := client.StreamInsert(ctx)
+		if err != nil {
+			t.Fatalf("error: %s", err)
+		}
+
+		wg := new(sync.WaitGroup)
+		wg.Add(1)
+		go func() {
+			defer wg.Done()
+
+			for {
+				result, err := stream.Recv()
+				if err != nil {
+					if err == io.EOF {
```

Comment:
I'm not sure, but should we use `errors.Is` to compare errors? 🤔 
ref: 
https://github.com/polyfloyd/go-errorlint#comparisons-of-errors

Reply:
`Recv()` will return `io.EOF` itself. We don't have to use `errors.Is`.